.github/workflows/ci-cd.yml

```diff
@@ -7,61 +7,155 @@ on:
   pull_request:
 
 jobs:
+  setup:
+    runs-on: ubuntu-latest
+    outputs:
+      short_sha: ${{ steps.sha.outputs.short_sha }}
+      go_version: ${{ steps.read.outputs.go_version }}
+      msys2_packages: ${{ steps.read.outputs.msys2_packages }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Generate a shortened SHA for artifacts
+        id: sha
+        run: echo "::set-output name=short_sha::$(git rev-parse --short HEAD)"
+      - name: Read build settings from repository
+        id: read
+        run: |
+          go_version=$(< .go-version)
+          echo "Go version: ${go_version}"
+          echo "::set-output name=go_version::${go_version}"
+          msys2_packages=$(tr -s '\n' ' ' < msys2-packages.txt)
+          echo "MSYS2 packages: ${msys2_packages}"
+          echo "::set-output name=msys2_packages::${msys2_packages}"
+
+  nebula:
+    name: Nebula
+    runs-on: ubuntu-20.04
+    needs: setup
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Setup Go toolchain
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ needs.setup.outputs.go_version }}
+
+      - name: Cache Go dependencies
+        uses: actions/cache@v2
+        with:
+          key: golang-${{ runner.os }}-${{ hashFiles('**/go.sum') }}
+          restore-keys: golang-${{ runner.os }}-
+          path: |
+            ~/go/pkg
+            ~/.cache/go-build
```

Comment:
I got this from `golangci-lint` and it seemed like there was no downside. It will only cache on the first compile with a set of deps, but that seems entirely acceptable.

Reply:
The downside is that cache space is limited to 5 GiB. However, I think we're only using about 1 GiB right now. With a few open PRs which change dependencies that could grow to 2-3 GiB but that still seems safe.
Unfortunately, there doesn't seem to be a way to check currently used cache space. I just look at the reported cache size in the cache restore steps.